alphastats/dataset/dataset.py

```diff
@@ -161,6 +150,37 @@ def _check_loader(loader):
                 "Invalid index_column: consider reloading your data with: AlphaPeptLoader, MaxQuantLoader, DIANNLoader, FragPipeLoader, SpectronautLoader"
             )
 
+    def _create_id_dicts(self, sep: str = ";") -> Tuple[dict, dict, dict]:
+        """Create mapprings from gene, protein to feature and from feature to repr."""
+
+        features = self.mat.columns.to_list()
+        gene_to_features_map = defaultdict(lambda: [])
+        protein_to_features_map = defaultdict(lambda: [])
+        feature_to_repr_map = defaultdict(lambda x: x)
+
+        for proteins, feature in self.rawinput[[Cols.INDEX, Cols.INDEX]].itertuples(
+            index=False
+        ):
+            if feature not in features:
+                continue
+            # TODO: Shorten list if too many ids e.g. to id1;...(19) if 20 ids are present
+            feature_to_repr_map[feature] = "ids:" + proteins
+            for protein in proteins.split(sep):
+                protein_to_features_map[protein].append(feature)
+
+        if Cols.GENE_NAMES in self.rawinput.columns:
+            for genes, feature in self.rawinput[
+                [Cols.GENE_NAMES, Cols.INDEX]
+            ].itertuples(index=False):
+                if feature not in features:
+                    continue
+                if isinstance(genes, str):
```

Comment:
what else could it be?

Reply:
Genes can be empty. features is updated based in what is left in mat after e.g. preprocessing.

Reply:
what kind of `instance` would `genes` then be?